packages/blade/src/components/TopNav/_decisions/decisions.md

```diff
@@ -33,16 +35,38 @@ The navigation bar is positioned at the top of the screen that provides quick ac
     <img src="razorpay-logo" />
   </TopNavBrand>
   <TopNavContent>
-    <TabNav>
-      <TabNavItem as={RouterLink} isActive href="/home" icon={HomeIcon} />
-      <TabNavItem as={RouterLink} href="/money">Money</TabNavItem>
-      <TabNavItem as={RouterLink} href="/payroll">Payroll</TabNavItem>
-      <Menu>
-        <TabNavItem trailingIcon={ChevronDown}>Others</TabNavItem>
-        <MenuOverlay>
-          { /* Other Menu Items */ }
-        </MenuOverlay>
-      </Menu>
+    <TabNav
+      items={[
+        { href: '/home', title: 'Home' },
+        { href: '/payroll', title: 'Payroll' },
+        { href: '/payments', title: 'Payments', isAlwaysOverflowing: true, },
+      ]}
+    >
+      {({ items, overflowingItems }) => {
+        return (
+          <>
+            <TabNavItems>
+              {items.map((item) => (
+                <TabNavItem href={item.href} title={title} />
+              ))}
+            </TabNavItems>
+            <Menu>
+              <TabNavItem title="More" />
+              {overflowingItems.map((item) => {
+                return (
+                  <MenuItem key={item.title}>
+                    <CustomExploreItem
+                      icon={item.icon}
+                      title={item.title}
+                      description={item.description}
+                    />
+                  </MenuItem>
+                );
+              })}
+            </Menu>
+          </>
+        );
+      }}
```

Comment:
Possible. 

https://github.com/razorpay/blade/blob/anu/tabnav-newapi/packages/blade/src/components/TopNav/_decisions/decisions.md#mobile-ux

Reply:
Okay so it's on the consumers to define what happens on mobile? 

Also what about the search on mobile? Are the consumers supposed to handle that as well?

The behaviour across viewports is well defined in the design, can we not leverage that as source of truth and handle the mobile variant accordingly from inside the component, without having the consumers define the behaviour?

Reply:
> Okay so it's on the consumers to define what happens on mobile?

Yes.

> Also what about the search on mobile? Are the consumers supposed to handle that as well?

The search input itself can be easily added as part of the TopNav or even in the page itself since we support slots.  

As to what happens on clicking the search it's product side decision, whether they want to open a modal/bottomsheet etc. This can be achieved on consumer end with existing components.